devicedefender/CMakeLists.txt

```diff
@@ -1,5 +1,25 @@
 cmake_minimum_required(VERSION 3.1)
-project(IotDeviceDefender-cpp CXX C)
+
+if (DEFINED SIMPLE_VERSION)
+    set(_sub_project_version ${SIMPLE_VERSION})
+else()
+    if (EXISTS "../VERSION")
+        file(READ "..//VERSION" version_simple)
+        string(STRIP ${version_simple} version_simple)
+        set(_sub_project_version ${version_simple})
+    endif()
+endif()
+
+if (DEFINED _sub_project_version)
+    message("Device Defender version is ${_sub_project_version}")
+    project(IotDeviceDefender-cpp
+            VERSION ${_sub_project_version}
+            LANGUAGES CXX)
+else()
+    message("Device Defender version is unknown")
+    project(IotDeviceDefender-cpp
+        LANGUAGES CXX)
```

Comment:
you can avoid redefining the project in two branches by defining `project()`and then later setting version via 
```
project(IotDeviceDefender-cpp
            LANGUAGES CXX)

if (DEFINED _sub_project_version)
    set_target_properties(IotDeviceDefender-cpp PROPERTIES VERSION ${_sub_project_version})
else()
    message("Device Defender version is unknown")
endif()

Reply:
But personally, given that they can't be built separately now, I'd kind of prefer we just try to minimize the amount of complex copy pasty code in each one